src/backends/perf_pt/collect.c

```diff
@@ -265,7 +265,19 @@ open_perf(pid_t target_tid) {
     attr.precise_ip = 3;
 
     // Acquire file descriptor through which to talk to Intel PT.
-    ret = syscall(SYS_perf_event_open, &attr, target_tid, -1, -1, 0);
+    //
+    // This syscall could return EBUSY, meaning another process or thread has
+    // locked the Perf device.
+    for (int tries = 500; tries > 0; tries--) {
+        printf("tries: %d\n", tries);
+        ret = syscall(SYS_perf_event_open, &attr, target_tid, -1, -1, 0);
+        if ((ret == -1) && (errno == EBUSY)) {
+            usleep(20); // Doesn't matter if usleep fails.
```

Comment:
Should we use `nanosleep`? My man page says `usleep is obsoleted` so I don't know if it's completely portable or not whereas `nanosleep` is in POSiX.

Reply:
Yes, even the Linux man page agrees:
```
       4.3BSD,  POSIX.1-2001.   POSIX.1-2001  declares this function obsolete;
       use nanosleep(2) instead.  POSIX.1-2008 removes  the  specification  of
       usleep().
```

Fix coming.